openmessaging-api/src/main/java/io/openmessaging/api/MessageBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2017 OpenMessaging
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.openmessaging.api;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * A message builder to build messages.
+ *
+ * <p>Use this class to create messages to be send by the OMS producer.
+ */
+public interface MessageBuilder {
+
+    /**
+     * Set the content of the message.
+     *
+     * @param data
+     *            array containing the payload
+     * @return message builder
+     */
+    default MessageBuilder setContent(byte[] data) {
```

Comment:
IMO, the default method is not suitable in our spec~

Reply:
any reason? it is a shortcut. but the implementation can still override it.